rfcs/0000-remove-builders.md

```diff
@@ -0,0 +1,34 @@
+
+The existing "Builder" relies on a polling mechanism to monitor an external Builder for buildpack and stack updates. External polling mechanisms in core kpack types have been a source of confusion that contradicts the pure declarative philosophy outlined in the [kpack monitors rfc](https://github.com/pivotal/kpack/pull/433).
+
+The "Builder's" limited control makes it difficult for users to actualize and understand kpack's functionality and make it a poor onboarding experience for new kpack community members. This discourages users from utilizing the extensible (Custom)Builder resource. Whereas onboarding with the explicit Stack, Store, Builder resources should make it easier to understand kpack's re-build logic. If we removed the existing Builder resource to invest in simplified CustomBuilder tooling and documentation we could focus on providing the best-unified experience for users and remove the toil that Builders create.
+
+Finally, for users to get the most value out of kpack they will eventually need to utilize the (Custom)Builder toolchain to leverage the granular control of buildpacks and stack images. For users to do that the cloud-native buildpacks community will need to have a robust ecosystem of buildpackages and stack images. We can encourage this community migration by discouraging or removing our existing Builder support.
+
+## Outcomes:
+- Reduce kpack onboarding confusion that results from the need to choose the right builder kind.
+- Encourage kpack users to utilize CustomBuilders which will provide the best kpack onboarding experience.
+- Simplify a future kpack cli from needing to use the terms `custom-cluster-builder` and `cluster-builder`.
+- Reduce the maintenance overhead of maintaining both sets of Builder resources. 
+
+## Actions to take:
+* Rename the CustomBuilder to Builder and ClusterCustomBuilder to ClusterBuilder and remove the Builder/ClusterBuilder resource.
+* Provide a kpack cli with functionality that initializes a set of ~~(Custom)~~Builders, Stacks, Stores to make onboarding and setup as simple as the previous builder experience.
+* We will consider recreating Builders in the eventual kpack monitors project.   
+
+## Complexity:
+There is already [active](https://github.com/pivotal/kpack/pull/437) [PRs](https://github.com/pivotal/kpack/pull/434) with major API renames. This is an ideal release window to include another major breaking api change.     
+
+ ## Prior Art:
+N/A 
+
+## Alternatives:
+
+The Builder could be renamed to something like "PreBuilt" Builder. This will allow users to continue utilizing external Builders. However, this approach does not resolve the underlying problems with the Builder concept and may continue to encourage users to use the PreBuiltBuilder.
+
+
+## Risks:
```

Comment:
This has been brought up by @djoyahoy – one risk is that the `Builder` is a [Cloud Native Buildpacks concept](https://buildpacks.io/docs/operator-guide/create-a-builder/) (however not a spec concept) and is an artifact that can be created and used by the Cloud Native Buildpacks `pack` cli project.

This means the builders created by `pack` can be used in `kpack` and makes builders a natural transition path from using `pack` cli to using `kpack`. Removing the `Builder` resource could make this transition tougher.

Reply:
If we do switch to renaming `CustomBuilder` > `Builder`, it would end up with the term `Builder` meaning 2 different things (as pointed out by Tyler above) in the CNB context and kpack context. And, considering kpack is a CNB platform implementation, this may not be ideal?

Reply:
> This means the builders created by pack can be used in kpack and makes builders a natural transition path from using pack cli to using kpack. Removing the Builder resource could make this transition tougher.

I am hopeful that our kpack cli will be able to help make the transition from pack to kpack smoother and easier for new users. 

I also would conjecture that it is more important to optimize for onboarding with (Custom)Builders in kpack than optimizing for onboarding in general.

Reply:
>If we do switch to renaming CustomBuilder > Builder, it would end up with the term Builder meaning two different things (as pointed out by @tylerphelan) in the CNB context and kpack context. And, considering kpack is a CNB platform implementation, this may not be ideal?

I'll need to think about this a bit more but, it is not quite two different things because the (Custom)Builder concept in kpack is broadly equivalent to `pack create-builder` and kpack (Custom)Builder image are usable as a builder with pack. So, it is more that pack created builders are not directly useable with kpack.